zerver/tests/test_auth_backends.py

```diff
@@ -1039,32 +1047,69 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
                                  parsed_url.path)
         self.assertTrue(uri.startswith('http://zulip.testserver/accounts/login/subdomain/'))
 
-        result = self.client_get(result.url)
+        if not user_avatar_url:
+            result = self.client_get(result.url)
+        else:
+            data['user_avatar_url'] = user_avatar_url
+            data['next'] = ''
+            token = store_login_data(data)
+            url_path = reverse('zerver.views.auth.log_into_subdomain', args=[token])
```

Comment:
we shouldn't be fiddling with these details - we should be verifying that the auth flow (executed by `social_auth_test` before calling this `stage_two_of_registration` function) did things correctly already.